mtags/src/main/scala/scala/meta/internal/pc/OverrideCompletions.scala

```diff
@@ -0,0 +1,489 @@
+
+      // infer necessary indent
+      //
+      // |object Test {
+      // |    class Foo extends Bar {} // inferred to 4
+      // |}
+      val lineStart = t.pos.source.lineToOffset(t.pos.line - 1)
+      val necessaryIndent = inferIndent(lineStart, text)
+
+      // infer indent for implementations
+      // if there's declaration in the class/object, follow its indent.
+      // otherwise the indent default to 2
+      val indent = typed.tpe.decls
+        .filter(sym =>
+          !sym.isSynthetic &&
+            !sym.isPrimaryConstructor &&
+            sym.pos.line != t.pos.line // filter out explicit primary constructor `class Foo(x: Int) ...`
+        )
+        .headOption
+        .map(existing => {
+          " " * inferIndent(
+            t.pos.source.lineToOffset(existing.pos.line - 1),
+            text
+          )
+        })
+        .getOrElse {
+          " " * (necessaryIndent + DefaultIndent)
+        }
+
+      val shouldCompleteBraces = !hasBody(text, t)
+
+      // if the both opening/closing braces located in a line:
+      // ```
+      // object {
+      //   class Foo extends Bar {}
+      // }
+      // ```
+      // or there's no body like this `class Foo extends Bar`.
+      // Add an newline and indent in the end of implementations, so that
+      // the closing brace is indented.
+      //
+      // object {
+      //   class Foo extends Bar {
+      //     override def foo = ???
+      //   }
+      // }
+      val lastIndent =
+        if (t.pos.source.offsetToLine(t.pos.start) ==
+            t.pos.source.offsetToLine(t.pos.end) || shouldCompleteBraces)
+          "\n" + " " * necessaryIndent
+        else ""
+
+      // Add opening/closing braces
+      // `object Foo extends Bar` to
+      // ```
+      // object Foo extends Bar {
+      //   override def method: Int = ???
+      // }
+      // ```
+      val start = if (shouldCompleteBraces) s" {\n${indent}" else s"\n${indent}"
+      val end = if (shouldCompleteBraces) s"${lastIndent}}" else lastIndent
+      val implementAll = new l.TextEdit(
+        range,
+        allAbstractEdits
+          .map(_.getNewText)
+          .mkString(
+            start,
+            s"\n${indent}",
+            end
+          )
+      )
+      implementAll :: allAbstractImports.toList
+    } else {
+      Nil
+    }
+  }
+
+  /**
+   * Get the position to insert implements for the given Template.
+   * `class Foo extends Bar {}` => retuning position would be right after the opening brace.
+   * `class Foo extends Bar` => retuning position would be right after `Bar`.
+   *
+   * @param text the text of the original source code.
+   * @param t the enclosing template for the class/object/trait we are implementing.
+   */
+  private def inferEditPosition(text: String, t: Template): Position = {
+    val start = t.pos.start
+    val end = t.pos.end
+    val offset = text.indexOf('{', start)
+    if (offset > 0 && offset < t.pos.end)
```

Comment:
```suggestion
    if (hasBody(text,t))
```
We could reuse logic from `hasBody`

Reply:
done https://github.com/scalameta/metals/pull/1379/commits/15d8e1feba9aed6cce18d9f6b74382f464b8be1e

Modified the `hasBody`'s return type to `Option[Int]` so that we can reuse `hasBody` from `inferEditPosition`.